mod/item.php

```diff
@@ -158,29 +158,32 @@ function item_post(App $a) {
 	// Now check that valid personal details have been provided
 	if (!can_write_wall($profile_uid) && !$allow_comment) {
 		notice(L10n::t('Permission denied.') . EOL) ;
+
 		if (x($_REQUEST, 'return')) {
 			goaway($return_path);
 		}
+
 		killme();
 	}
 
-
-	// is this an edited post?
-
+	// Init post instance
 	$orig_post = null;
 
-	if ($post_id) {
+	// is this an edited post?
+	if ($post_id > 0) {
 		$orig_post = Item::selectFirst(Item::ITEM_FIELDLIST, ['id' => $post_id]);
 	}
 
 	$user = dba::selectFirst('user', [], ['uid' => $profile_uid]);
+
 	if (!DBM::is_result($user) && !$parent) {
 		return;
 	}
 
 	$categories = '';
+	$postopts = '';
 
-	if ($orig_post) {
+	if (is_array($orig_post)) {
```

Comment:
You could use `!empty()` instead, being an array wouldn't ensure the following array keys to be set.

Reply:
The initialization happens with `null`. Yeah, paradigms ... ;-) Okay, I change it.